accounts/abi/bind/precompiletemplate.go

```diff
@@ -0,0 +1,366 @@
+res, err := {{$contract.Type}}ABI.UnpackInput("{{$method.Original.Name}}", input)
+if err != nil {
+	return nil, err
+}
+unpacked := *abi.ConvertType(res[0], new({{bindtype $input.Type $structs}})).(*{{bindtype $input.Type $structs}})
+return unpacked, nil
+}
+
+// Pack{{.Normalized.Name}} packs [{{decapitalise $input.Name}}] of type {{bindtype $input.Type $structs}} into the appropriate arguments for {{$method.Original.Name}}.
+// the packed bytes include selector (first 4 func signature bytes).
+// This function is mostly used for tests.
+func Pack{{$method.Normalized.Name}}( {{decapitalise $input.Name}} {{bindtype $input.Type $structs}},) ([]byte, error) {
+	return {{$contract.Type}}ABI.Pack("{{$method.Original.Name}}", {{decapitalise $input.Name}},)
+}
+{{else}}
+// Pack{{.Normalized.Name}} packs the include selector (first 4 func signature bytes).
+// This function is mostly used for tests.
+func Pack{{.Normalized.Name}}() ([]byte, error) {
+	return {{$contract.Type}}ABI.Pack("{{.Original.Name}}")
+}
+{{end}}
+
+{{if len .Normalized.Outputs | lt 1}}
+// Pack{{capitalise .Normalized.Name}}Output attempts to pack given [outputStruct] of type {{capitalise .Normalized.Name}}Output
+// to conform the ABI outputs.
+func Pack{{capitalise .Normalized.Name}}Output (outputStruct {{capitalise .Normalized.Name}}Output) ([]byte, error) {
+	return {{$contract.Type}}ABI.PackOutput("{{.Original.Name}}",
+		{{- range .Normalized.Outputs}}
+		outputStruct.{{capitalise .Name}},
+		{{- end}}
+	)
+}
+
+{{else if len .Normalized.Outputs | eq 1 }}
+{{$method := .}}
+{{$input := index $method.Normalized.Outputs 0}}
+// Pack{{capitalise .Normalized.Name}}Output attempts to pack given {{decapitalise $input.Name}} of type {{bindtype $input.Type $structs}}
+// to conform the ABI outputs.
+func Pack{{$method.Normalized.Name}}Output ({{decapitalise $input.Name}} {{bindtype $input.Type $structs}}) ([]byte, error) {
+	return {{$contract.Type}}ABI.PackOutput("{{$method.Original.Name}}", {{decapitalise $input.Name}})
+}
+{{end}}
+
+func {{decapitalise .Normalized.Name}}(accessibleState PrecompileAccessibleState, caller common.Address, addr common.Address, input []byte, suppliedGas uint64, readOnly bool) (ret []byte, remainingGas uint64, err error) {
+	if remainingGas, err = deductGas(suppliedGas, {{.Normalized.Name}}GasCost); err != nil {
+		return nil, 0, err
+	}
+
+	{{- if not .Original.IsConstant}}
+	if readOnly {
+		return nil, remainingGas, vmerrs.ErrWriteProtection
+	}
+ 	{{- end}}
+
+	{{- if len .Normalized.Inputs | eq 0}}
+	// no input provided for this function
+	{{else}}
+	// attempts to unpack [input] into the arguments to the {{.Normalized.Name}}Input.
+	// Assumes that [input] does not include selector
+	// You can use unpacked [inputStruct] variable in your code
+	inputStruct, err := Unpack{{capitalise .Normalized.Name}}Input(input)
+	if err != nil{
+		return nil, remainingGas, err
+	}
```

Comment:
Can we include something below here so that there's no complaint about `inputStruct` being unused?

Perhaps:

```
_ = inputStruct // CUSTOM CODE OPERATES ON INPUT
```